thalassa/normalization.py

```diff
@@ -120,37 +145,59 @@ def normalize_generic(ds: xarray.Dataset) -> xarray.Dataset:
 def normalize_schism(ds: xarray.Dataset) -> xarray.Dataset:
     ds = ds.rename(
         {
-            "nSCHISM_hgrid_edge": "edge",
-            "nSCHISM_hgrid_face": "face",
-            "nSCHISM_hgrid_node": "node",
-            "SCHISM_hgrid_face_nodes": "face_nodes",
-            "nMaxSCHISM_hgrid_face_nodes": "max_no_vertices",
-            "SCHISM_hgrid_node_x": "lon",
-            "SCHISM_hgrid_node_y": "lat",
+            "nSCHISM_hgrid_edge": EDGE_DIM,
+            "nSCHISM_hgrid_face": FACE_DIM,
+            "nSCHISM_hgrid_node": NODE_DIM,
+            "SCHISM_hgrid_face_nodes": CONNECTIVITY,
+            "nMaxSCHISM_hgrid_face_nodes": VERTICE_DIM,
+            "SCHISM_hgrid_node_x": X_DIM,
+            "SCHISM_hgrid_node_y": Y_DIM,
         },
     )
     if "nSCHISM_vgrid_layers" in ds.dims:
         # I.e. OLD Schism IO or "merged" new IO
         ds = ds.rename(
             {
-                "nSCHISM_vgrid_layers": "layer",
+                "nSCHISM_vgrid_layers": VERTICAL_DIM,
             },
         )
     # SCHISM output uses one-based indices for `face_nodes`
     # Let's ensure that we use zero-based indices everywhere.
-    ds["face_nodes"] -= 1
+    ds[CONNECTIVITY] -= 1
+    return ds
+
+
+def normalize_telemac(ds: xarray.Dataset) -> xarray.Dataset:
+    ds = ds.rename(
+        {
+            "node": NODE_DIM,
+            "x": X_DIM,
+            "y": Y_DIM,
+        },
+    )
+    if "plan" in ds.dims:
+        # I.e. OLD Schism IO or "merged" new IO
+        ds = ds.rename(
+            {
+                "plan": VERTICAL_DIM,
+            },
+        )
+
+    # TELEMAC output uses one-based indices for `face_nodes`
+    # Let's ensure that we use zero-based indices everywhere.
+    ds[CONNECTIVITY] = ((FACE_DIM, VERTICE_DIM), ds.attrs['ikle2'] - 1)
```

Comment:
This is most probably xarray-selafin related, but why is `ikle2` an attribute and not a variable?

Reply:
Luc actually shifted `ikle2` and `ikle3` from coordinates to attributes because it removed useless dimensions.
Here's the [commit](https://github.com/seareport/xarray-selafin/pull/17#issuecomment-1973673429). 
We might want to move them back in the variables at some point I agree.

Reply:
I wouldn't call the connectivity useless. Furthermore, it might be that serializing attributes needs more space compared to data variables (lack of compression etc). Haven't tested it though.